kola/tests/kubernetes/kubelet.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform/conf"
+	"github.com/coreos/mantle/util"
+)
+
+var (
+	staticPodConfig = conf.Ignition(`{
+    "ignition": {
+        "config": {},
+        "security": {
+            "tls": {}
+        },
+        "timeouts": {},
+        "version": "2.2.0"
+    },
+    "networkd": {},
+    "passwd": {},
+    "storage": {
+        "files": [
+            {
+                "contents": {
+                    "source": "data:,kind%3A%20KubeletConfiguration%0AapiVersion%3A%20kubelet.config.k8s.io%2Fv1beta1%0AclusterDomain%3A%20kube.local%0AclusterDNS%3A%20%5B10.254.0.10%5D%0AcgroupDriver%3A%20systemd%0AstaticPodPath%3A%20%2Fetc%2Fkubernetes%2Fmanifests%2F%0Aauthorization%3A%0A%20%20mode%3A%20AlwaysAllow%0Aauthentication%3A%0A%20%20anonymous%3A%0A%20%20%20%20enabled%3A%20true%0A%20%20webhook%3A%0A%20%20%20%20enabled%3A%20false%0A",
+                    "verification": {}
+                },
+                "filesystem": "root",
+                "mode": 420,
+                "path": "/etc/kubernetes/kubeletconfig"
+            },
+            {
+                "contents": {
+                    "source": "data:,apiVersion%3A%20v1%0Akind%3A%20Pod%0Ametadata%3A%0A%20%20name%3A%20static-web%0A%20%20labels%3A%0A%20%20%20%20role%3A%20myrole%0Aspec%3A%0A%20%20containers%3A%0A%20%20%20%20-%20name%3A%20web%0A%20%20%20%20%20%20image%3A%20nginx%0A%20%20%20%20%20%20ports%3A%0A%20%20%20%20%20%20%20%20-%20name%3A%20web%0A%20%20%20%20%20%20%20%20%20%20containerPort%3A%2080%0A%20%20%20%20%20%20%20%20%20%20protocol%3A%20TCP%0A",
+                    "verification": {}
+                },
+                "filesystem": "root",
+                "mode": 420,
+                "path": "/etc/kubernetes/manifests/static-pod.yaml"
+            }
+        ]
+    },
+    "systemd": {
+        "units": [
+            {
+                "contents": "[Unit]\nDescription=Kubernetes Kubelet\nRequires=docker.service\nAfter=docker.service\n\n[Service]\nExecStart=/usr/bin/hyperkube \\\n    kubelet \\\n        --config /etc/kubernetes/kubeletconfig\n\nRestart=always\nRestartSec=10\n\n[Install]\nWantedBy=multi-user.target\n",
```

Comment:
Just a note, this and below assume kubelet is using docker (and that the host has docker at all) - so it'll need some adaption for Fedora CoreOS at least, and shortly RHCOS isn't going to use Docker either.  But for now we can do this, just let's keep in mind it'll (hopefully shortly) need to change.

Reply:
I reduced the verification down to a curl command that doesn't depend on the the container runtime.  I left docker in the kubelet unit file though.